karapace/kafka/types.py

```diff
@@ -0,0 +1,55 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+
+from confluent_kafka import (
+    Message,
+    TIMESTAMP_CREATE_TIME,
+    TIMESTAMP_LOG_APPEND_TIME,
+    TIMESTAMP_NOT_AVAILABLE,
+    TopicPartition,
+)
+from confluent_kafka.admin import (
+    BrokerMetadata,
+    ClusterMetadata,
+    ConfigResource,
+    ConfigSource,
+    NewTopic,
+    OffsetSpec,
+    PartitionMetadata,
+    ResourceType,
+    TopicMetadata,
+)
+from confluent_kafka.error import KafkaError, KafkaException
+
+import enum
+
+__all__ = (
```

Comment:
I'm trying to make `karapace.kafka` as self-contained as possible and avoiding `confluent_kafka` references leaking anywhere else.

Having `kafka` and `aiokafka` spread all over the codebase has been a pain in gradually replacing them. Also if/when we add/use more functionality from `confluent_kafka` it's possible that we'll need to wrap some of its classes with our own to enhance them - it's going to be easier to do so if things are in one place.

Reply:
I like that idea

Reply:
I don't agree with this, and I doubt the reason that kafka-python and aiokafka are hard to replace is because we import directly from them rather from an aliasing namespace like this. In fact, I think it would have been harder because in order to follow the pattern during the transitioning phase, entities from both third-party libraries would have been forced to compete for names in the same namespace.

In order for this to actually have the intended effect, we would need to introduce our own abstraction layer and explicitly implement adapters for each backend. As we don't plan to support multiple backends in the long term, I don't really see any value in this though and I think it's OK that we tie ourselves into the libraries we're using, even if it makes it a bit harder to replace them.

As there are two voices in the opposite direction from mine here, I see where this is leaning. Please consider this one more round and then mark this as resolved if that's where you're still landing 🙂

Reply:
Thanks for challenging this! I'll remove the re-exports and change back to the previous state where we use `confluent_kafka` imports throughout.

While I do prefer re-exports here and hiding `confluent_kafka` elsewhere (from a structural point of view), I prefer future-proofing more. Given this structure, using types and classes from `confluent_kafka` in other places in the codebase would have to be a 2-step process always - slightly inconvenient with the inevitable mixing of the two styles as direct imports from the library is the path of least resistance. 👍 

Will resolve the comment once the change is done.